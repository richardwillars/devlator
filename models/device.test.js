const deviceModule = require('./device');

describe('models/device', () => {
  it('should export the schema', () => {
    const mongoose = {
      model: () => {},
      Schema: class Schema {},
    };
    const deviceModel = deviceModule(mongoose);
    expect(deviceModel.schema).toMatchSnapshot();
  });

  it('should export the model', () => {
    const modelMock = {};
    const mongoose = {
      model: jest.fn().mockReturnValue(modelMock),
      Schema: class Schema {
        constructor(props) {
          this.props = props;
        }
      },
    };
    const deviceModel = deviceModule(mongoose);

    expect(deviceModel.model).toEqual(modelMock);
    expect(mongoose.model).toHaveBeenCalledTimes(1);
    expect(mongoose.model.mock.calls[0][0]).toEqual('Device');
    const initialisedSchema = mongoose.model.mock.calls[0][1];
    expect(initialisedSchema instanceof mongoose.Schema).toEqual(true);
    expect(initialisedSchema.props).toMatchSnapshot();
  });
});
