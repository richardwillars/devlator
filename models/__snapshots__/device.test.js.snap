// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/device should export the model 1`] = `
Object {
  "_id": Object {
    "required": true,
    "type": [Function],
    "unique": true,
  },
  "created": Object {
    "default": [Function],
    "required": false,
    "type": [Function],
  },
  "driver": Object {
    "required": true,
    "type": [Function],
  },
  "specs": Object {
    "default": Object {},
    "required": true,
    "type": [Function],
  },
  "type": Object {
    "required": true,
    "type": [Function],
  },
}
`;

exports[`models/device should export the schema 1`] = `
Object {
  "_id": Object {
    "required": true,
    "type": [Function],
    "unique": true,
  },
  "created": Object {
    "default": [Function],
    "required": false,
    "type": [Function],
  },
  "driver": Object {
    "required": true,
    "type": [Function],
  },
  "specs": Object {
    "default": Object {},
    "required": true,
    "type": [Function],
  },
  "type": Object {
    "required": true,
    "type": [Function],
  },
}
`;
